src/ess_rhat.jl

```diff
@@ -239,19 +257,184 @@ The ESS and ``\\widehat{R}`` values can be computed for the following estimators
     doi: [10.1214/20-BA1221](https://doi.org/10.1214/20-BA1221)
     arXiv: [1903.08008](https://arxiv.org/abs/1903.08008)
 """
-function ess_rhat(samples::AbstractArray{<:Union{Missing,Real},3}; kwargs...)
-    return ess_rhat(Statistics.mean, samples; kwargs...)
+function ess(samples::AbstractArray{<:Union{Missing,Real},3}; kind=:bulk, kwargs...)
+    if kind === :bulk
+        return _ess(Val(:bulk), samples; kwargs...)
+    elseif kind === :tail
+        return _ess(Val(:tail), samples; kwargs...)
+    elseif kind === :basic
+        return _ess(Val(:basic), samples; kwargs...)
+    elseif kind isa Symbol
+        throw(ArgumentError("the `kind` `$kind` is not supported by `ess`"))
+    else
+        return _ess(kind, samples; kwargs...)
+    end
 end
-function ess_rhat(f, samples::AbstractArray{<:Union{Missing,Real},3}; kwargs...)
-    x = _expectand_proxy(f, samples)
+function _ess(estimator, samples::AbstractArray{<:Union{Missing,Real},3}; kwargs...)
+    x = _expectand_proxy(estimator, samples)
     if x === nothing
-        throw(ArgumentError("the estimator $f is not yet supported by `ess_rhat`"))
+        throw(ArgumentError("the estimator $estimator is not yet supported by `ess`"))
```

Comment:
Makes me wonder, would it be more natural to add a default fallback for `_expectand_proxy` that does not return `nothing` but throws the descriptive error message directly?

Reply:
Since the error includes the function name, it would seem odd to me to put that error in `_expectand_proxy`, and the traceback would then point to this internal function not relevant to the user. Since this function is meant only for internal use, I don't think it's a problem for it to have this fallback.

Reply:
The second argument is slightly weak though, I think, since also currently the traceback points to an internal function.

Reply:
True, but `_ess` is more similar to `ess` than  `_expectand_proxy` is.